CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -254,19 +245,30 @@ public function buildQuickForm() {
     $this->assign('component', $this->_component);
     $this->assign('email', $this->_contributorEmail);
 
-    $this->add('text', 'fee_amount', ts('Fee Amount'),
-      $attributes['fee_amount']
-    );
-    $this->addRule('fee_amount', ts('Please enter a valid monetary value for Fee Amount.'), 'money');
-
-    $this->add('text', 'net_amount', ts('Net Amount'),
-      $attributes['net_amount']
-    );
-    $this->addRule('net_amount', ts('Please enter a valid monetary value for Net Amount.'), 'money');
-
     $js = NULL;
+    // render backoffice payment fields only on offline mode
     if (!$this->_mode) {
       $js = array('onclick' => "return verify( );");
+
+      $this->add('select', 'payment_instrument_id',
```

Comment:
Ya but I think why shouldn't render these quickForm elements needlessly when its not in use.. So I thought why not to move the code here rather exposing it.